lib/gush/client.rb

```diff
@@ -122,7 +123,7 @@ def find_job_class(job_name)
     end
 
     def find_job(workflow_id, job_name)
-      job_name_match = /(?<klass>\w*[^-])-(?<identifier>.*)/.match(job_name)
+      job_name_match = /(?<klass>.*)\|(?<identifier>.*)/.match(job_name)
```

Comment:
Why the separator between the klass and the identifier has changed from `-` to `|`?

Reply:
Seems a quite all bug.
[Upstream](https://github.com/chaps-io/gush/blob/master/spec/gush/job_spec.rb#L109), this is a `|` separator.

Debugging the lib yesterday show this code path is never called… :thinking: 
This is quite strange, job key is [`class_name`](https://github.com/chaps-io/gush/blob/master/lib/gush/client.rb#L115), the ID is only use inside the redis set (and is the only key…).
So there is no possible `class_name-ID` nor `class_name|ID` on redis.
Seems a quick way to manually retrieve jobs from redis without doing 2 call manually.

Reply:
This is this kind of code I want to look into and perform some cleaning.
There is some other strange thinks in the code base.